File: mario.py
height = 0
rows = 0
cols = 0
count = 0
maxrows = -1
while maxrows < 1 or maxrows > 20:
    try:
        maxrows = int(input("What is the height(maximum number of rows) in your pyramid? "))
        count += 1
        if count >= 1 and maxrows < 1 or maxrows > 20:
            print("Your entry is out of range(1-20).")
    except ValueError:
        print("You must enter an integer(a number) between 1 and 20")


print("The height of your pyramid(maximum number of rows) is: {}".format(maxrows))
#The code below prints the rows for BOTH first and second triangle
for rows in range(1, maxrows + 1):
    #The code below pads the first triangle with spaces before the *'s.
    for spaces in range(maxrows - cols, -1, -1):
        print(end=" ")
    for cols in range(1, rows + 1):
        print("*", end="")
    print(end="  ")
    #The code below prints the columns of the second triangle
    for cols in range(1, rows + 1):
        print("*", end="")
    #The next line prevents all the stars from being printed on the same line.
    print("\n")

File: mario.c
//Troy Kelley - CS50 - Week 1
#include <stdio.h>

int main(void) {
int maxrows;
int rows;
int colstars;
int colspaces;
int counter = 0;

do {
    printf("What is the number of rows in your pyramid?? ");
    scanf("%d", &maxrows);
    counter += 1;
/*Next line makes sure that input's been requested at least once (otherwise the error message is shown before the user      even responds to the first question!) It also checks that the input is in  range.*/
    if (counter >= 1 && ((maxrows < 1) || (maxrows > 8))) {
        printf("Your entry is out of range(1-8) \n");}
    }
while (!((maxrows >=1) && (maxrows <= 8)));

//The next line prints the rows of BOTH triangles.
for (rows = 1; rows <= maxrows; rows++) {
    for (colspaces = maxrows - rows; colspaces >= 0; colspaces--){
        printf(" ");
    }
    //The line below finishes the printing (the columns) of the //    //first triangle.
    for (colstars = 1; colstars < rows + 1; colstars++){
        printf("#");
    }
    //The next (printf) line is what creates the gap of 2 spaces 
    //between the first and second triangles.
    printf("  ");
    //The next line finishes the printing (the columns) of the
    //second triangle. 
    for (colstars = 1; colstars <= rows; colstars++) 
    {
        printf("#");
    }
    {
    //The next line is necessary, otherwise all the stars will be
    //printed on the same line.
    printf("\n");
  
   
}
}}
